File: how-to-GCD.py
def gcd(x, y = 1):
	xdiv = []
	ydiv = []
	for i in range(1, x + 1):
		if x % i == 0:
			xdiv.append(i)
	for j in range(1, y + 1):
		if y % j == 0:
			ydiv.append(j)
	retgcd = 0
	for k in xdiv:
		if (k in ydiv):
			retgcd = k
	return(retgcd)
print(gcd(10, 15))

File: how-to-GCD.cpp
#include <iostream>
#include <vector>
using namespace std;
int gcd(int x, int y = 1){
    std::vector<int> xdiv;
    std::vector<int> ydiv;
    for (auto i=1;i<=x;i++) if (x % i == 0) xdiv.push_back(i);
    for (auto j=1;j<=y;j++) if (y % j == 0) ydiv.push_back(j);
    int retgcd = 0;
    auto find = [](std::vector<int> vec, int x){int ct = 0;for (int i : vec) if (i == x) ct ++;if (ct > 0) return true;else return false;};
    for (int k : xdiv) if (find(ydiv, k)) retgcd = k;
    return retgcd;
}
int main(){
    std::cout << gcd(50, 75) << std::endl;
}
